core/src/main/scala/org/apache/spark/deploy/worker/ExecutorRunner.scala

```diff
@@ -135,7 +135,7 @@ private[spark] class ExecutorRunner(
       logInfo("Launch command: " + command.mkString("\"", "\" \"", "\""))
 
       builder.directory(executorDir)
-      builder.environment.put("SPARK_LOCAL_DIRS", appLocalDirs.mkString(","))
+      builder.environment.put("SPARK_EXECUTOR_DIRS", appLocalDirs.mkString(File.pathSeparator))
```

Comment:
Oh right, this is `path.separator` not `file.separator`, `:` not `/`. 

Yes that's exactly the fix that needs to be made.

`SPARK_EXECUTOR_DIRS` is a new system property right? where is it read? I am missing something.